### PR TITLE
Compiler Version support for Visual Studion 2017+

### LINKED ISSUE #1612: add support for compiler version selection
The current MSVC has a bug that makes it impossible to work with for me. Visual Studio has a setting called "MSVC Toolset Version" which internally sets the <VCToolsVersion> tag.  
Note that this is not the Platform Toolset but a way to select a specific version of the selected Platform Toolset.

I would like to set the msvc toolset version used for a specific platform toolset. I can imagine that this feature also applies to other project types as well.

Even more convenient would be a compiler blacklisting feature to blacklist compiler versions known to not work. But that would obviously require premake to know which compiler versions are installed.

### PR DESCRIPTION
**Pull request by nickclark2016**
**What does this PR do?**

Adds compiler version support for VS 2017 and newer

**How does this PR change Premake's behavior?**

No breaking changes, all new functionality.

Syntax
```lua
toolsversion "1.2.3"
```

This will add the following to the globals group in VS 2017 and newer project files.
```xml
<VCToolsVersion>1.2.3</VCToolsVersion>
```

It is worth noting that this does **no** checks for legal tool versions. That is on the user. I believe it would be more hassle than it is worth to go check for all installed tool version, then validate user input against that.

**Anything else we should know?**

Closes #1612

**Did you check all the boxes?**

- [x] Focus on a single fix or feature; remove any unrelated formatting or code changes
- [x] Add unit tests showing fix or feature works; all tests pass
- [x] Mention any [related issues](https://github.com/premake/premake-core/issues) (put `closes #XXXX` in comment to auto-close issue when PR is merged)
- [x] Follow our [coding conventions](https://github.com/premake/premake-core/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] Minimize the number of commits
- [x] Align [documentation](https://github.com/premake/premake-core/tree/master/website) to your changes

*You can now [support Premake on our OpenCollective](https://opencollective.com/premake). Your contributions help us spend more time responding to requests like these!*

### CODE REVIEW

File: modules/vstudio/vs2010_vcxproj.lua [hunk 1]
```diff
@@ -1699,11 +1702,11 @@
...
 		elseif p.languages.iscpp(cfg.compileas) then
 			m.element("CompileAs", condition, "CompileAsCpp")
-		elseif cfg.compileas == "Module" then
+		elseif cfg.compileas == "Module" then
 			m.element("CompileAs", condition, "CompileAsCppModule")
-		elseif cfg.compileas == "ModulePartition" then
+		elseif cfg.compileas == "ModulePartition" then
 			m.element("CompileAs", condition, "CompileAsCppModuleInternalPartition")
-		elseif cfg.compileas == "HeaderUnit" then
+		elseif cfg.compileas == "HeaderUnit" then
```
Comment (nickclark2016) [inline modules/vstudio/vs2010_vcxproj.lua:1709]: Looks like this got caught in light ending fixes.

File: src/_premake_init.lua [hunk 2]
```diff
@@ -185,9 +185,9 @@
 			"C++",
 			"Objective-C",
 			"Objective-C++",
-			"Module",
-			"ModulePartition",
-			"HeaderUnit"
+			"Module",
+			"ModulePartition",
+			"HeaderUnit"
```
Comment (nickclark2016) [inline src/_premake_init.lua:190]: Same as above. This was caught in line ending fixes.

**Review by starkos — APPROVED** [review]
Nice, thanks!

**Comment by KyrietS** [comment]
Hey @nickclark2016, you forgot to add `"toolsversion"` to `website/sidebars.js` file. Can you fix it in another commit? 

Btw. the latest version of docusaurus introduced some kind of automation for sidebars. It's not perfect yet but I'll try this feature in time.

**Comment by nickclark2016** [comment]
@KyrietS Ahh crap, yup, I'll do that.

**Comment by KyrietS** [comment]
@nickclark2016 \*cough\* \*cough\* 🙃

**Comment by nickclark2016** [comment]
> 
> 
> @nickclark2016 *cough* *cough* 🙃

Oh crap, totally slipped my mind. Standby

**Pull request merged**